gcc/rust/parse/rust-parse.h

```diff
@@ -86,7 +86,7 @@ struct ParseRestrictions
 // TODO: if updated to C++20, ManagedTokenSource would be useful as a concept
 template <typename ManagedTokenSource> class Parser
 {
-private:
+public:
```

Comment:
Maybe keep this as private and move `peek_current_token()` in public members? I don't know how much it affects the rest of the codebase to leak these private functions

Reply:
yeah lets do that, i will need to do this fix manually to make public the parse functions we want

Reply:
fixed